src/main.zig

```diff
@@ -1676,8 +1678,7 @@ fn buildOutputType(
                                     const key = linker_arg[0..equals_pos];
                                     const value = linker_arg[equals_pos + 1 ..];
                                     if (mem.eql(u8, key, "build-id")) {
```

Comment:
```suggestion
                                    if (mem.eql(u8, key, "--build-id")) {
```

also see #15457

Reply:
The goal of this PR was to implement a minimally viable implementation for build-id that supports styles without breaking existing behavior, so the flag would need to be duplicated.

#15459 is better fit for this, since it cleans up all the interfaces.